packages/dds/tree/src/test/simple-tree/tree.spec.ts

```diff
@@ -247,8 +247,7 @@ describe("object allocation tests", () => {
 		assert.equal(countAfter, countBefore + 1);
 	});
 
-	// TODO: AB#8575 re-enable this test once leaf access on arrays does not allocate flex nodes
-	it.skip("accessing leaf on array node does not allocate flex nodes", () => {
+	it("accessing leaf on array node does not allocate flex nodes", () => {
```

Comment:
with this test written the way it is, its hard to tell if the extra allocation is from the field, or the leaf. If you put two leaves in the array and access then both, then its clear that if there is only a single extra flex tree entity, its the field.

That said, I think allocating the flex tree entity for the field is fine for arrays Afterall its just once for the array, and we use it for lots of stuff), so maybe make an array of two items, and read the first one before you set countBefore so it includes the field, then read the second on and check that does not allocate more.

Reply:
Makes sense, I'll update it.